### ISSUE TITLE
Improve backend tests (especially ln-gateway) scalability

### ISSUE THREAD

**Issue by dpc** [fedimint/fedimint#3209]
devimint starts only one lnd gateway. LND gateway can have only one HTLC interceptor at the time. All ln gateway tests want to be an interceptor. Tests try to run in parallel. Effectively ln-gateway only work because they keep trying until they register.

In addition some tests will deadlock if run in parallel in certain combinations because of this.

We also have some tests that use [`lock_exclusive`](https://github.com/fedimint/fedimint/blob/19606001b703dd44a39bebdb2ceb623115528e83/fedimint-testing/src/btc/mod.rs#L13) to get an exclusive lock on bitcoin node.

This is all not very scalable.

Some approaches we could take (possibly in combination) sorted by amount of work:

#### Add LND lock

Implement something like `lock_exclusive` for LND instance, to at least eliminate deadlocks.

### Start multiple LND instances

Combined with the locking should be relatively simple and unblock some level of parallelism.

### Have devimint start and manage a pool of LND nodes

Just have each test request an LND instance to use, then get it locked, all managed by devimint. Would require some layer of rpc from test to devimint, but can be very elastic.

Some way to cap the maximum amount of these might be needed.

### Have a devimint manage a whole pool of environments.

Like "pool of LND nodes" but for the whole environment. This could eliminate bitcoind lock_exclusive. If a test needs an exclusive access to bitcoind, just give it a whole deviming instance. So we could have one devimin binary spawn multiple "devimint env" instances.

Some way to cap the maximum amount of these might be needed.

**Comment by dpc** [comment]
@fedimint/infra

**Comment by m1sterc001guy** [comment]
@bradleystachurski I assigned this to you, let me know if that's ok or if you need any pointers.

**Comment by bradleystachurski** [comment]
@m1sterc001guy Thanks! Definitely ok, I'll holler if I get stuck and need some pointers.

**Comment by bradleystachurski** [comment]
Dev call: split into separate issue

**Comment by elsirion** [comment]
@dpc @bradleystachurski still relevant?

**Comment by m1sterc001guy** [comment]
IMO this can be closed. `ln-gateway` tests were refactored to only use mocks, we rely on devimint tests now for E2E which can be parallelized much easier.
